pepper-apis/src/main/java/org/broadinstitute/ddp/filter/RateLimitFilter.java

```diff
@@ -0,0 +1,39 @@
+package org.broadinstitute.ddp.filter;
+
+import java.util.Map;
+import javax.servlet.ServletException;
+
+import org.broadinstitute.ddp.exception.DDPException;
+import org.eclipse.jetty.servlets.DoSFilter;
+import spark.Filter;
+import spark.Request;
+import spark.Response;
+
+public class RateLimitFilter implements Filter {
+
+    private final int maxRequestsPerSecond;
+
+    private final int burst;
+
+    private final DoSFilter rateLimitFilter;
+
+    public RateLimitFilter(int maxRequestsPerSecond, int burst) {
+        this.maxRequestsPerSecond = maxRequestsPerSecond;
+        this.burst = burst;
+        DoSFilter rateLimitFilter = new DoSFilter();
+        rateLimitFilter.setEnabled(true);
+        try {
+            rateLimitFilter.init(new FilterConfig("DOSFilter", Map.of("delayMs", "0",
```

Comment:
Hmm, I did some more local testing. It doesn't seem like DoSFilter has the same "burst" as nginx.

Let's assume `maxRequestsPerSec` is 10 and `burst` is 15. If we set `delayMs=-1`, then it only accepts 10 and rejects all the rest. If we set `delayMs=0`, then it accepts all the requests (sends 10, throttles all the rest) so no requests get rejected even though its over the limit. In nginx, it treats it as a "burst" and will handle 10 + 15 = 25 requests (10 send to server, 15 queued), and all the rest will be rejected with `429`.

I'm not sure what's the best configuration here. Maybe we set a high `maxRequestsPerSec` and `delayMs=-1`? What do folks think?

Reply:
Our chattiest services so far are the autocompleters for physician/institution, mailing address, and the patches for activity instance answers.  I don't see these fire more than a few times per second, so I'd advocate for keeping `maxRequestsPerSec=10` and `delayMs=-1`.  How's that sound?

Reply:
I think that sounds good. Maybe bump it up just a bit higher for `prod`?